crates/flux-fhir-analysis/src/lib.rs

```diff
@@ -207,10 +229,33 @@ fn assoc_refinement_def(
 
     if let Some(assoc_reft) = assoc_reft {
         let wfckresults = genv.check_wf(impl_id)?;
-        Ok(rty::EarlyBinder(conv::conv_assoc_reft_def(genv, assoc_reft, &wfckresults)?))
+        Ok(Some(rty::EarlyBinder(conv::conv_assoc_reft_def(genv, assoc_reft, &wfckresults)?)))
     } else {
-        Err(QueryErr::InvalidAssocReft { impl_id: impl_id.to_def_id(), name })?
+        Ok(None)
+    }
+}
+
+fn assoc_refinement_def(
+    genv: GlobalEnv,
+    impl_id: LocalDefId,
+    name: Symbol,
+) -> QueryResult<rty::EarlyBinder<rty::Lambda>> {
+    let err = Err(QueryErr::InvalidAssocReft { impl_id: impl_id.to_def_id(), name });
```

Comment:
Same here. the very first thing should be `let impl_id = genv.maybe_extern_id(impl_id)`. I also thing the `impl_id` should be the resolved id.